drivers/driver_functions.go

```diff
@@ -0,0 +1,93 @@
+package drivers
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher/v2"
+	"github.com/rancher/rancher-auth-service/util"
+	"time"
+)
+
+//WebhookDriver interface for all drivers
+type WebhookDriver interface {
+	ConstructPayload(input map[string]interface{}) (map[string]interface{}, error)
+	Execute(payload map[string]interface{}) (int, map[string]interface{}, error)
+}
+
+type ServiceScaler struct {
+	Id string
```

Comment:
What is this Id for? Where is it used? Also, Id is not the right format for golang and it will not pass gofmt. The right format is ID. You should run `make ci` and make sure the build is completing successfully and that all gofmt, golint, and govet errors are addressed.